core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -331,9 +331,21 @@ private String newTableMetadataFilePath(TableMetadata meta, int newVersion) {
     return metadataFileLocation(meta, String.format("%05d-%s%s", newVersion, UUID.randomUUID(), fileExtension));
   }
 
+  /**
+   * Parse the version from table metadata file name.
+   *
+   * @param metadataLocation table metadata file location
+   * @return version of the table metadata file in success case and
+   * -1 if the version is not parsable (as a sign that the metadata is not part of this catalog)
+   */
   private static int parseVersion(String metadataLocation) {
     int versionStart = metadataLocation.lastIndexOf('/') + 1; // if '/' isn't found, this will be 0
     int versionEnd = metadataLocation.indexOf('-', versionStart);
+    if (versionEnd < 0) {
+      LOG.warn("Found filesystem table's metadata : {}", metadataLocation);
```

Comment:
I don't think this is a good thing to warn about. Warnings are for potential problems, while this is a normal case if you convert a table or control a name some other way.

This is also inaccurate. We can't really know why the file didn't include a version and shouldn't guess.

Reply:
removed the log warning and added it as a code comment.

Reply:
The more we talk about this, the less convinced I am that we should differentiate between the parsing errors. Nothing really prevents a developer to create their own `TableOperations` and using their own metadata filename format.

I think it would be good to have a log for the event in any case, at least in info level.

Maybe rewrite the whole parsing part to this:
```
  private static int parseVersion(String metadataLocation) {
    try {
      int versionStart = metadataLocation.lastIndexOf('/') + 1; // if '/' isn't found, this will be 0
      int versionEnd = metadataLocation.indexOf('-', versionStart);
      return Integer.valueOf(metadataLocation.substring(versionStart, versionEnd));
    } catch (Throwable e) {
      LOG.info("Unable to parse version from metadata location: {}", metadataLocation, e);
      return -1;
    }
  }
```

What do you think?

Reply:
SGTM. waiting for @rdblue 's opinion to avoid me working back and forth on this.

Reply:
I think this should check for `versionEnd`. That indicates that we're using a file with no version and should return -1.

If we also want to return -1 for `NumberFormatException` and warn, I'm all for that. But we should definitely not catch `Throwable`. That's dangerous. Let's handle the cases we know about.

Reply:
Thanks for the inputs @rdblue. Agree that catching `Throwable` is not a good practice. I have reverted the latest change. So, final code is same as what you described above.